pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -262,6 +262,16 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 			if defaultCluster == nil {
 				continue
 			}
+
+			// if the service uses persistent sessions, override status allows
+			// DRAINING endpoints to be kept as 'UNHEALTHY' coarse status in envoy.
+			// Will not be used for normal traffic, only when explicit override.
+			if service.Attributes.Labels[features.PersistentSessionLabel] != "" {
+				defaultCluster.cluster.CommonLbConfig.OverrideHostStatus = &core.HealthStatusSet{
+					Statuses: []core.HealthStatus{core.HealthStatus_HEALTHY, core.HealthStatus_UNHEALTHY, core.HealthStatus_DRAINING},
```

Comment:
Default is `UNKNOWN, HEALTHY, DEGRADED`... do we want to include DEGRADED as well just to be safe? probably doesn't matter though since we never set it

Reply:
I think this looks good.  Only thing for general Envoy we can add is put terminating pods status as automatically DRAINING and that would help stateful sessions (instead of just looking draining label)